### PR TITLE
TPI-3216: Move CC to Invoice

### PR DESCRIPTION
**Pull request by candrasaputra**
## Description

`put the description of your PR here`

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Cosmetic change (text changing or color changing)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code (if necessary), particularly in hard-to-understand areas
- [x] My changes generate no new vulnerabilities
- [ ] I have added tests that prove my fix is effective or that my feature works

## Screenshot
- Checkout
Screen Shot 2021-01-27 at 07 48 18
Screen Shot 2021-01-27 at 07 46 06

- Refund
Screen Shot 2021-01-27 at 07 42 54
Screen Shot 2021-01-27 at 07 45 08

### CODE REVIEW

File: Xendit/M2Invoice/Model/Payment/CC.php [hunk 1]
```diff
@@ -2,141 +2,22 @@
...
-    protected $_code = self::CODE;
-    protected $_minAmount = 5000;
-    protected $_maxAmount = 200000000;
-    protected $methodCode = 'CCFORM';
-
-    protected $_isGateway = true;
-    protected $_canAuthorize = true;
-    protected $_canCapture = true;
+    protected $_code = 'cc';
+    protected $_minAmount = 10000;
```
Comment (hakiemaul) [inline Xendit/M2Invoice/Model/Payment/CC.php:13]: we can preserve old min value i think

File: Xendit/M2Invoice/Model/Payment/CC.php [hunk 2]
```diff
@@ -2,141 +2,22 @@
...
+    protected $_maxAmount = 1000000000;
```
Comment (hakiemaul) [inline Xendit/M2Invoice/Model/Payment/CC.php:14]: also old max value

**Review by hakiemaul — APPROVED** [review]

File: Xendit/M2Invoice/view/frontend/web/js/view/payment/method-renderer/cc.js [hunk 3]
```diff
@@ -161,127 +53,27 @@ define(
...
+                if (!billingAddress) {
+                    self.messageContainer.addErrorMessage({'message': 'Please enter your billing address'});
+                    return false;
                 }
 
-                if (!Xendit.card.validateExpiry(tokenData.card_exp_month, tokenData.card_exp_year)) {
-					this.showError('The card expiry that you entered does not meet the expected format. Please try again by entering the 2 digits of the month (MM) and the last 2 digits of the year (YY). Code: 200031')
-					return false;
+                if (totals.grand_total < 10000) {
+                    self.messageContainer.addErrorMessage({'message': 'Xendit doesn\'t support purchases less than Rp 10,000.'});
```
Comment (IreneGohtami) [inline Xendit/M2Invoice/view/frontend/web/js/view/payment/method-renderer/cc.js:72]: we can use the standardized error message for this validation

File: Xendit/M2Invoice/Helper/Data.php [hunk 4]
```diff
@@ -262,7 +257,7 @@ public function xenditPaymentMethod( $payment ){
         
         //method name => frontend routing
         $listPayment = [
-            "cc" => "cc",
+            "cc" => "credit_card",
             "cchosted" => "cchosted",
```
Comment (IreneGohtami) [inline Xendit/M2Invoice/Helper/Data.php:261]: i think we no longer have cchosted right?

**Review by yanoandri — APPROVED** [review]

**Review by hakiemaul — APPROVED** [review]

**Pull request merged**